File: app/src/main/java/mohtasham/paydar/sabalan/ezbazi/model/Game.java
package mohtasham.paydar.sabalan.ezbazi.model;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

import mohtasham.paydar.sabalan.ezbazi.model.common.Photo;
import mohtasham.paydar.sabalan.ezbazi.model.common.Video;

public class Game {
  private int id;
  private int game_info_id;
  private int price;
  private int is_pending;
  private String name;
  private int console_type_id;
  private int age_class;
  private String production_date;
  private String region;
  private int can_play_online;
  private String company_name;
  private String description;
  private ArrayList<Photo> photos;
  private ArrayList<Video> videos;
  private ArrayList<String> genres;
  private String console_name;

  public static class Parser{

    public static Game parse(JSONObject jsonObject){
      Game game = new Game();
      try {
        JSONObject game_info_obj = new JSONObject();
        game_info_obj = jsonObject.getJSONObject("game_info");

        game.setId(jsonObject.getInt("id"));
        game.setGame_info_id(jsonObject.getInt("game_info_id"));
        game.setPrice(jsonObject.getInt("price"));
        game.setIs_pending(jsonObject.getInt("is_pending"));

        //game_info
        game.setName(game_info_obj.getString("name"));
        game.setConsole_type_id(game_info_obj.getInt("console_type_id"));
        game.setAge_class(game_info_obj.getInt("age_class"));
        game.setProduction_date(game_info_obj.getString("production_date"));
        game.setRegion(game_info_obj.getString("region"));
        game.setCan_play_online(game_info_obj.getInt("can_play_online"));
        game.setCompany_name(game_info_obj.getString("company_name"));
        game.setDescription(game_info_obj.getString("description"));
        JSONArray photoArray = game_info_obj.getJSONArray("photos");
        ArrayList<Photo> photos = new ArrayList<>();
       for (int i=0 ; i<photoArray.length() ; i++){
         Photo photo;
         photo = Photo.Parser.parse(photoArray.getJSONObject(i));
         photos.add(photo);
       }
       game.setPhotos(photos);

        JSONArray videoArray = game_info_obj.getJSONArray("videos");
        ArrayList<Video> videos = new ArrayList<>();
        for (int i=0 ; i<videoArray.length() ; i++){
          Video video;
          video = Video.Parser.parse(videoArray.getJSONObject(i));
          videos.add(video);
        }
        game.setVideos(videos);

        JSONArray genreArray = game_info_obj.getJSONArray("genres");
        ArrayList<String> genres = new ArrayList<>();
        for (int i=0 ; i<genreArray.length() ; i++){
          String genre;
          genre = genreArray.getJSONObject(i).getString("name");
          genres.add(genre);
        }
        game.setGenres(genres);

        game.setConsole_name(game_info_obj.getJSONObject("console").getString("name"));

      } catch (JSONException e) {
        e.printStackTrace();
      }
      return game;
    }
  }





  public String getConsole_name() {
    return console_name;
  }

  public void setConsole_name(String console_name) {
    this.console_name = console_name;
  }

  public int getId() {
    return id;
  }

  public void setId(int id) {
    this.id = id;
  }

  public int getPrice() {
    return price;
  }

  public void setPrice(int price) {
    this.price = price;
  }

  public int getGame_info_id() {
    return game_info_id;
  }

  public void setGame_info_id(int game_info_id) {
    this.game_info_id = game_info_id;
  }


  public int isIs_pending() {
    return is_pending;
  }

  public void setIs_pending(int is_pending) {
    this.is_pending = is_pending;
  }


  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public int getConsole_type_id() {
    return console_type_id;
  }

  public void setConsole_type_id(int console_type_id) {
    this.console_type_id = console_type_id;
  }

  public int getAge_class() {
    return age_class;
  }

  public void setAge_class(int age_class) {
    this.age_class = age_class;
  }

  public String getProduction_date() {
    return production_date;
  }

  public void setProduction_date(String production_date) {
    this.production_date = production_date;
  }

  public String getRegion() {
    return region;
  }

  public void setRegion(String region) {
    this.region = region;
  }

  public int isCan_play_online() {
    return can_play_online;
  }

  public void setCan_play_online(int can_play_online) {
    this.can_play_online = can_play_online;
  }

  public String getCompany_name() {
    return company_name;
  }

  public void setCompany_name(String company_name) {
    this.company_name = company_name;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public ArrayList<Photo> getPhotos() {
    return photos;
  }

  public void setPhotos(ArrayList<Photo> photos) {
    this.photos = photos;
  }

  public ArrayList<Video> getVideos() {
    return videos;
  }

  public void setVideos(ArrayList<Video> videos) {
    this.videos = videos;
  }


  public int getIs_pending() {
    return is_pending;
  }

  public int getCan_play_online() {
    return can_play_online;
  }

  public ArrayList<String> getGenres() {
    return genres;
  }

  public void setGenres(ArrayList<String> genres) {
    this.genres = genres;
  }


}

File: app/src/main/java/sabalan/paydar/mohtasham/ezibazi/model/Game.kt
package sabalan.paydar.mohtasham.ezibazi.model

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

import java.util.ArrayList

class Game {
    var id: Int = 0
    var game_info_id: Int = 0
    var price: Int = 0
    private var is_pending: Int = 0
    var name: String? = null
    var console_type_id: Int = 0
    var age_class: Int = 0
    var production_date: String? = null
    var region: String? = null
    var count: Int = 0
    private var can_play_online: Int = 0
    var company_name: String? = null
    var description: String? = null
    var photos: ArrayList<Photo>? = null
    var videos: ArrayList<Video>? = null
    var genres: ArrayList<String>? = null
    var console_name: String? = null
    var app_main_photos_url: ArrayList<String>? = null
    var app_cover_photo_url: String? = null

    object Parser {

        fun parse(jsonObject: JSONObject): Game {
            val game = Game()
            try {
                var game_info_obj = JSONObject()
                game_info_obj = jsonObject.getJSONObject("game_info")

                game.id = jsonObject.getInt("id")
                game.game_info_id = jsonObject.getInt("game_info_id")
                game.price = jsonObject.getInt("price")
                game.setIs_pending(jsonObject.getInt("is_pending"))
                game.region = jsonObject.getString("region")
                game.count = jsonObject.getInt("count")

                //game_info
                game.name = game_info_obj.getString("name")
                game.console_type_id = game_info_obj.getInt("console_type_id")
                game.age_class = game_info_obj.getInt("age_class")
                game.production_date = game_info_obj.getString("production_date")
                //        game.setRegion(game_info_obj.getString("region"));
                game.setCan_play_online(game_info_obj.getInt("can_play_online"))
                game.company_name = game_info_obj.getString("company_name")
                game.description = game_info_obj.getString("description")
                val photoArray = game_info_obj.getJSONArray("photos")
                val photos = ArrayList<Photo>()

                val main_photo_urls = ArrayList<String>()
                for (i in 0 until photoArray.length()) {
                    val photo: Photo
                    photo = Photo.Parser.parse(photoArray.getJSONObject(i))
                    photos.add(photo)

                    if (photo.type!!.contains("app_main")) {
                        main_photo_urls.add(photo.url)
                    } else if (photo.type!!.contains("app_cover")) {
                        game.app_cover_photo_url = photo.url
                    }

                }
                game.app_main_photos_url = main_photo_urls
                game.photos = photos

                val videoArray = game_info_obj.getJSONArray("videos")
                val videos = ArrayList<Video>()
                for (i in 0 until videoArray.length()) {
                    val video: Video
                    video = Video.Parser.parse(videoArray.getJSONObject(i))
                    videos.add(video)
                }
                game.videos = videos

                val genreArray = game_info_obj.getJSONArray("genres")
                val genres = ArrayList<String>()
                for (i in 0 until genreArray.length()) {
                    val genre: String
                    genre = genreArray.getJSONObject(i).getString("name")
                    genres.add(genre)
                }
                game.genres = genres

                game.console_name = game_info_obj.getJSONObject("console").getString("name")

            } catch (e: JSONException) {
                e.printStackTrace()
            }

            return game
        }
    }


    fun isIs_pending(): Int {
        return is_pending
    }

    fun setIs_pending(is_pending: Int) {
        this.is_pending = is_pending
    }

    fun isCan_play_online(): Int {
        return can_play_online
    }

    fun setCan_play_online(can_play_online: Int) {
        this.can_play_online = can_play_online
    }


    fun getIs_pending(): Int {
        return is_pending
    }

    fun getCan_play_online(): Int {
        return can_play_online
    }


}
